### PR TITLE
Transform anything that isn't a character

### PR DESCRIPTION
**Pull request by MoOx**
This way, something like a slash in a filepath are transformed too.

Didn't make any tests in the commit since I pushed this change from github interface (lazy commit).

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -11,7 +11,13 @@ module.exports = function dashify(str) {
   if (typeof str !== 'string') {
     throw new TypeError('dashify expects a string.');
   }
-  str = str.replace(/[A-Z]/g, '-$&');
```
Comment (jonschlinkert) [inline index.js]: 4 of the 5 existing tests will fail with this change lol.

File: index.js [hunk 2]
```diff
@@ -11,7 +11,13 @@ module.exports = function dashify(str) {
...
-  str = str.replace(/[ \t]/g, '-');
```
Comment (jonschlinkert) [inline index.js]: here, all you need to do is change the regex to `/[ \t\W]/g`

File: index.js [hunk 3]
```diff
@@ -11,7 +11,13 @@ module.exports = function dashify(str) {
...
-  return str.toLowerCase();
+  
+  // dashify
+  str = str.toLowerCase();
+  str = str.replace(/\W+/g, '-');
+  // trim dash
+  str = str.replace(/^-+/, '');
+  str = str.replace(/-+$/, '');
```
Comment (jonschlinkert) [inline index.js]: to trim end dashes, you can do `/^-+|-+$/g` so it's accomplished with one regex.

**Comment by jonschlinkert** [comment]
If you want to add unit tests and make the suggested changes so existing tests pass I'd be happy to merge. I think the changes would be good

File: index.js (hunk 3 above)
Comment (MoOx) [inline index.js]: Nope. This doesn't work.

**Pull request closed without merging**